wezterm-gui/src/glyphcache.rs

```diff
@@ -889,20 +889,21 @@ impl<T: Texture2d> GlyphCache<T> {
                 let pixel = buffer.pixel_mut(x, y);
                 let (current, _, _, _) = SrgbaPixel::with_srgba_u32(*pixel).as_rgba();
                 let value = current.saturating_add(val);
-                *pixel = SrgbaPixel::rgba(value, value, value, 0xff).as_srgba32();
+                *pixel = SrgbaPixel::rgba(0xFF, 0xFF, 0xFF, val).as_srgba32();
             }
 
             for x in 0..self.metrics.cell_size.width as usize {
-                let vertical = wave_height as f32 * (x as f32 * x_factor).cos();
-                let v1 = vertical.floor();
-                let v2 = vertical.ceil();
+                let vertical = wave_height as f32 / -3. * (x as f32 * x_factor).sin()
+                    + wave_height as f32 / 2.;
+                let v = vertical.round();
 
                 for row in 0..self.metrics.underline_height as usize {
-                    let value = (255. * (vertical - v1).abs()) as u8;
-                    add(x, row + y + v1 as usize, 255 - value, max_y, buffer);
-                    add(x, row + y + v2 as usize, value, max_y, buffer);
+                    let value = (255. * (vertical - v).abs()) as u8;
+                    add(x, row + y + v as usize, 0x88, max_y, buffer);
```

Comment:
```suggestion
                    add(x, row + y + v as usize, value, max_y, buffer);
```

passing value should give some amount a grey-level control and an opportunity for AA.

I think the original secondary `add` call might be good to restore now that you've found that the alpha value wasn't respected in the shader.  That second `add` with `255 - value` appears to be the thing that computes the anti-aliased pixel value.